src/monmove.cpp

```diff
@@ -341,7 +342,10 @@ void monster::plan( const mfactions &factions )
     } else if( friendly < 0 && sees( g->u ) ) {
         if( rl_dist( pos(), g->u.pos() ) > 2 ) {
             set_dest( g->u.pos() );
-        } else {
+        }
+        else if( has_flag( MF_EGGLAYING ) && has_effect( effect_laid_egg ) ) {
```

Comment:
Just that the two checks should be in `on_idle`, not here, in the outside.
If you're having problems because of the `unset_dest` below, move it into `on_idle`.

So `on_idle` would check if the monster is an egg layer and then try to find a nest for it, but for all other monsters it would just unset destination. It would also unset destination for egg layers who don't manage to find a nest (or don't want to, because of egg timer).